src/PDPVerifier.sol

```diff
@@ -388,6 +384,8 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         uint256 challengeEpoch = nextChallengeEpoch[setId];
         require(block.number >= challengeEpoch, "premature proof");
         require(proofs.length > 0, "empty proof");
+        require(challengeEpoch != NO_CHALLENGE_SCHEDULED, "no challenge scheduled");
+        require(proofSetOwner[setId] == msg.sender, "only the owner can prove possession");
```

Comment:
@ZenGround0 This check already exists above and is a duplicate.

Reply:
Thanks